lib/publiccloud/instance.pm

```diff
@@ -477,4 +503,178 @@ sub network_speed_test() {
+
+    return $ret;
+}
+
+
+=head2 store_boottime_db
+
+    store_boottime_db();
+
+Save data collected with measure_boottime in a DB;
+Mainly stored on a remote InfluxDB on a Grafana server.
+
+=cut
+
+sub store_boottime_db() {
+    my ($self, $results) = @_;
+    return unless (get_var('_PUBLIC_CLOUD_PERF_PUSH_DATA') && $results);
+
+    my $url = get_var('PUBLIC_CLOUD_PERF_DB_URI');
+    my $token = get_var('_PUBLIC_CLOUD_PERF_DB_TOKEN');
+    unless ($url && $token) {
+        record_info("WARN", "PUBLIC_CLOUD_PERF_DB_URI or _PUBLIC_CLOUD_PERF_DB_TOKEN is missing ", result => 'fail');
+        return 0;
+    }
+
+    my $org = get_var('PUBLIC_CLOUD_PERF_DB_ORG', 'qec');
+    my $db = get_var('PUBLIC_CLOUD_PERF_DB', 'perf');
+
+    my $tags = {
+        instance_type => get_required_var('PUBLIC_CLOUD_INSTANCE_TYPE'),
+        job_id => get_current_job_id(),
+        os_provider => get_required_var('PUBLIC_CLOUD_PROVIDER'),
+        os_build => get_required_var('BUILD'),
+        os_flavor => get_required_var('FLAVOR'),
+        os_version => get_required_var('VERSION'),
+        os_kernel_release => $results->{kernel_release},
+        os_kernel_version => $results->{kernel_version},
+    };
+
+    $tags->{os_pc_build} = get_var('PUBLIC_CLOUD_QAM') ? 'N/A' : get_var('PUBLIC_CLOUD_BUILD', 0);
+    $tags->{os_pc_kiwi_build} = get_var('PUBLIC_CLOUD_QAM') ? 'N/A' : get_var('PUBLIC_CLOUD_BUILD_KIWI', 0);
+
+    record_info("STORE analyze", 'bootup');
+    # Store values in influx-db
+
+    my $data = {
+        table => 'bootup',
+        tags => $tags,
+        values => $results->{analyze}
+    };
+    influxdb_push_data($url, $db, $org, $token, $data);
+
+    record_info("STORE blame", $results->{type});
+    $tags->{boottype} = $results->{type};
+    $data = {
+        table => 'bootup_blame',
+        tags => $tags,
+        values => $results->{blame}
+    };
+    influxdb_push_data($url, $db, $org, $token, $data);
+}
+
+sub systemd_time_to_second
+{
+    my $str_time = trim(shift);
+
+    if ($str_time !~ /^(?<check_min>(?<min>\d{1,2})\s*min\s*)?((?<sec>\d{1,2}\.\d{1,3})s|(?<ms>\d+)ms)$/) {
+        record_info("WARN", "Unable to parse systemd time '$str_time'", result => 'fail');
+        return -1;
+    }
+    my $sec = $+{sec} // $+{ms} / 1000;
+    $sec += $+{min} * 60 if (defined($+{check_min}));
+    return $sec;
+}
+
+sub extract_analyze_time {
+    my $str_time = shift;
+    my $res = {};
+    ($str_time) = split(/\r?\n/, $str_time, 2);
+    $str_time =~ s/Startup finished in\s*//;
+    $str_time =~ s/=(.+)$/+$1 (overall)/;
+    for my $time (split(/\s*\+\s*/, $str_time)) {
+        $time = trim($time);
+        my ($time, $type) = $time =~ /^(.+)\s*\((\w+)\)$/;
+        $res->{$type} = systemd_time_to_second($time);
+        return 0 if ($res->{$type} == -1);
+    }
+    map { return 0 unless exists($res->{$_}) } qw(kernel initrd userspace overall);
```

Comment:
as on L#548, I consider the usage of `map` as rather a bit odd to me.
Those usages are rather a check, whereas `map` is mostly handy to manipulate data.
IMHO I would rather go for a classic loop.

Reply:
Sorry I  don't agree: why to increase the code, instead of reducing it using languages smart solutions? `map` isn't any virus :)  and the loop increases the code complexity for a simple log file transfer.

Reply:
As more clarification, this new code is a change, in order to satisfy a request to upload each log file, instead of the previous  *.tar.gz containing all the files

Reply:
I am not against `map`, it is definitely a powerful feature, but it feels like a misuse as you could not really fully benefit from it here.
Also, code wise, loop is just more readable here as well. But that is a matter of taste ofc

Reply:
I don't have strong opinion here ... I would say that I am closer to Martin's position that map is misused in this case . But also I don't see any importance in this decision . Whether it will be map or for loop does not change much . So let's use golden rule **"the one who do decides"** and leave it to @m-dati  . Deal ?

Reply:
No objects, it just got my attention, therefore I have raised the point

Reply:
Sorry, I am lerner, not sure to get the point of _map misuse_ here, also looking in [documentation](https://perldoc.perl.org/functions/map) 
> map evaluates the BLOCK ...  for each element of LIST ... and composes a list of the results ...",

anyway I will change the code this way:
```
foreach (@LIST) { BLOCK };
```

Reply:
done